src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/Razor/ExtractToNewComponentCodeActionProviderTest.cs

```diff
@@ -0,0 +1,232 @@
+            Context = new VSInternalCodeActionContext()
+        };
+
+        var location = new SourceLocation(cursorPosition, -1, -1);
+        var context = CreateRazorCodeActionContext(request, location, documentPath, contents);
+        context.CodeDocument.SetFileKind(FileKinds.Legacy);
+
+        var provider = new ExtractToNewComponentCodeActionProvider(LoggerFactory);
+
+        // Act
+        var commandOrCodeActionContainer = await provider.ProvideAsync(context, default);
+
+        // Assert
+        Assert.Null(commandOrCodeActionContainer);
+    }
+
+    [Fact]
+    public async Task Handle_InProperMarkup_ReturnsNull()
+    {
+        // Arrange
+        var documentPath = "c:/Test.razor";
+        var contents = """
+            page "/"
+            
+            <PageTitle>Home</PageTitle>
+            
+            <div id="parent">
+                <div>
+                    <h1>Div a title</h1>
+                    <p>Div $$a par</p>
+                </div>
+                <div>
+                    <h1>Div b title</h1>
+                    <p>Div b par</p>
+                </div>
+            </div
+            
+            <h1>Hello, world!</h1>
+            
+            Welcome to your new app.
+            """;
+        TestFileMarkupParser.GetPosition(contents, out contents, out var cursorPosition);
+
+        var request = new VSCodeActionParams()
+        {
+            TextDocument = new VSTextDocumentIdentifier { Uri = new Uri(documentPath) },
+            Range = new Range(),
+            Context = new VSInternalCodeActionContext()
+        };
+
+        var location = new SourceLocation(cursorPosition, -1, -1);
+        var context = CreateRazorCodeActionContext(request, location, documentPath, contents);
+
+        var provider = new ExtractToNewComponentCodeActionProvider(LoggerFactory);
+
+        // Act
+        var commandOrCodeActionContainer = await provider.ProvideAsync(context, default);
+
+        // Assert
+        Assert.Null(commandOrCodeActionContainer);
+    }
+
+    // Holding off on this test until configured correctly (fails on CI)
+    //[Theory]
+    //[InlineData("""
+    //<div id="parent">
+    //    [|<div>
+    //        <h1>Div a title</h1>
+    //        <p>Div a par</p>
+    //    </div>|]
+    //    <div>
+    //        <h1>Div b title</h1>
+    //        <p>Div b par</p>
+    //    </div>
+    //</div>
+    //""")]
+    //[InlineData("""
+    //<div id="parent">
+    //    <div>
+    //        <h1>Div a title</h1>
+    //        [|<p>Div a par</p>|]
+    //    </div>
+    //    <div>
+    //        <h1>Div b title</h1>
+    //        <p>Div b par</p>
+    //    </div>
+    //</div>
+    //""")]
+    //[InlineData("""
+    //<div id="parent">
+    //    <div>
+    //        <h1>Div a title</h1>
+    //        [|<p>Div a par</p>
+    //    </div>
+    //    <div>
+    //        <h1>Div b title</h1>
+    //        <p>Div b par</p>|]
+    //    </div>
+    //</div>
+    //""")]
```

Comment:
This `[InlineData]` is far too bulky, IMO. Please consider one of the following:

1. Switch over to `[MemberData]`. There are plenty of examples in the repo of using `[MemberData]` rather than `[InlineData]`.
2. Create three separate `[Fact]` methods that call this code with different data. (You'd want to switch this to a private method in this case.) @davidwengier has been writing a lot of tests in this style for co-hosting. Here's an [example](https://github.com/dotnet/razor/blob/main/src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostUriPresentationEndpointTest.cs).

For tests like this one, I'll often go with the second approach. It can be helpful when a test fails to clearly see what it's input was, without having to go dig through an array of member data.